main.go

```diff
@@ -105,18 +107,17 @@ func main() {
 		os.Exit(1)
 	}
 
-	iamClientAndRegionFactory := func(session awsclientgo.ConfigProvider) (iamiface.IAMAPI, string) {
-		client := awsiam.New(session)
-		return client, client.SigningRegion
+	iamClientFactory := func(session awsclientgo.ConfigProvider) iamiface.IAMAPI {
+		return awsiam.New(session)
```

Comment:
IAM is a global service so it always returns a region `us-east-1` so using that value was useless, rather decided to pass the region from the CR